ug: Team-based access to devices and device groups (#501)
foundriesio/docs: source/user-guide/account-management/team-based-access.rst

A Team is comprised of:

 * **Access control**: what :ref:`scopes <ref-scopes>` are granted to the team, e.g. ``source:read``.

 * **Members**: what users belong to the team.

 * **Device groups**: what device groups belong to the team.

> Once a user is assigned a team, their access will be limited to the scopes granted to that team.

Reviewer: ```suggestion
Once a Factory member is assigned a team, their access will be limited to the scopes granted to that team.
```

Author: I haven't changed this sentence, this is how it is in the current document.

> Also, a user will get access to the team's device groups and their devices in accordance with the team scopes.

Reviewer: ```suggestion
Also, a member will get access to the team's device groups and their devices in accordance with the team scopes.
```

Author: "member" - is it factory member or team member? I think in this context it should be either "user" or "factory member".

   * :ref:`API Scopes <ref-scopes>` for available scopes.

.. _Access to Device Groups:

Team Based Access to Device Groups
----------------------------------

> By default, a user can access only those device groups that are created by the user.

Reviewer: ```suggestion
By default, a member can access only those device groups that they created.
```

Author: I think it should be a user in this context too. If a user is in a team (member) then it is not the "default" case.

> Also, the user can access devices that are owned by them or devices that are in device groups created by the user.

Reviewer: ```suggestion
Related, a member can access devices they own, or devices that are in device groups they created.
```

Author: The same, just a user.

> A factory admin can grant a user with access to any device groups and their devices.

Reviewer: ```suggestion
A Factory admin can grant a member access to any device groups and their devices.
```

Author: The "user" I meant in this context is not necessary a team member. An admin must add a user to a team to grant them access to devices/groups.

Reviewer 2: I guess @kprosise was meaning that "member" on this page can mean one of two things:
- a factory member;
- a team member.

And IIUC the documentation team is OK with that ambiguity in scope of this page.

> A factory admin can grant a user with access to any device groups and their devices.

Reviewer 2: 1. s/factory/Factory/
2. s/any device group/Factory device group/

Maybe @kprosise would add more to this.

> To do so, an admin should add a device group to a team that a user is member of.

Reviewer: ```suggestion
To do so, an admin should add a device group to a team that a member belongs to.
```

Author: I think I should add one more step to make it clear to user. An admin should add a user to a team prior to adding a group to the team.

Or, as an option, we can assume that "user" in this guide is already a team member, so we can reference "user" as "member" as @kprosise suggested.

@doanac how do you recon?

Should we just write on a top something like: "It is assumed that a user is a team member" or "adding a user to a team is prerequisite"? Or maybe, just use "member" term everywhere as Katrina suggested.

> The actions that a user can perform over device groups and their devices are determined by  ``devices:read``, ``devices:read-update``, and ``devices:delete`` scopes set for a team.

Reviewer 2: s/<space><space>/<space>/

> The actions that a user can perform over device groups and their devices are determined by  ``devices:read``, ``devices:read-update``, and ``devices:delete`` scopes set for a team.

Reviewer: ```suggestion
The actions that a member can perform over device groups and their devices are determined by the ``devices:read``, ``devices:read-update``, and ``devices:delete`` scopes set for a team.
```

> The actions that a user can perform over device groups and their devices are determined by  ``devices:read``, ``devices:read-update``, and ``devices:delete`` scopes set for a team.

Reviewer 2: I think that exact scope names and their meaning are already defined elsewhere.

So, this could be simplified to just `determined by scopes set for a team`; maybe also write that text into a link to other section.

Reviewer 3: We can link this as a note: https://docs.foundries.io/latest/reference-manual/factory/api-access.html?highlight=tokens#token-scopes

Author: I just wanted to emphasize which of the scopes are relevant to the access to devices/groups.

Reviewer 2: Just IMHO less text is usually better. By this time the reader should already familiarize with our scopes and what they mean.
If they did not manage to do that - they'd anyway contact the customer support.
Thus repeating this information here again does not help.

Example
^^^^^^^

A Factory has two teams in place and one device group, ``test-lab-devices``.

Members of the "read-only-users" team can "see" everything, but can not make changes.

> Also, the members have a read-only access to the `test-lab-devices` group and its devices since the group is included to the team.

Reviewer 2: 1. `s/the members/members/` or `s/the members/they/` (if you refer to the same members as in the first sentence).
2. Also, in the first sentence it is vague what does `"see" everything"` means. Is it all resources, or just group and its devices?
    Maybe, this whole thing could be written in just one sentence, or using a bullet list.

Reviewer: A table may be helpful to summarize

Author: The "see" is a term used in the current document version, in the first example, so I decided to continue with the same style :).

Reviewer 2: That's OK, I understand what "see" means here; but I don't understand what "everything" means.

.. figure:: /_static/userguide/account-management/team-with-group-and-read-access.png
   :align: center

>    :alt: Read-only team with a device group

Reviewer 3: ```suggestion
   :alt: "read-only-users" scopes: read-only team with a device group 
```

Author: done

The "lab-dev-users" team includes ``devices:read-update`` scope.

> Therefore, members of this team can modify the `test-lab-devices` group and its devices.

Reviewer 3: ```suggestion
Therefore, members of this team can modify the ``test-lab-devices`` group and its devices.
```

Reviewer 2: JFYI @mike-sul this was my confusion as well.

Everywhere `text` means special inline formatting; but not for Sphinx - there it is italic font. Special formatting is ``text`` as Vanessa suggested.

.. figure:: /_static/userguide/account-management/team-with-group-and-write-access.png
   :align: center

>    :alt: Read-only team with a device group

Reviewer 3: ```suggestion
   :alt: "lab-dev-users" scopes: read-update team with a device group
```

Author: done
